### PR TITLE
Apps skip links on newslettersignup

### PR DESCRIPTION
**Pull request by dblatcher**
## What does this change?
Adds a skip link around the newsletter promotion blocks in AR, similar to https://github.com/guardian/dotcom-rendering/pull/6950 for DCR.

### testing
To test locally, you will need to bypass the `getBridgetVersion` function; the dev server does not emulate Bridget, so the function will hang and timeout when running locally. To replicate a version of the app with a version that does support the newsletter sign-up embeds, change the function as below: 
```typescript 
       export const getBridgetVersion = async (): Promise<Optional<string>> => {
       return Optional.some('5.0.0')

	try {
		return await environmentClient
			.nativeThriftPackageVersion()
			.then((result) => Optional.some(result));
	} catch (error) {
		console.log('nativeThriftPackageVersion', { error });
		return Optional.none();
	}
};
```
To see the behaviour for a version of the app when the sign-ups are not support, change the '5.0.0' to '1.0.0'

Run the dev server and open an article with a `promotedNewsletter` eg:
http://localhost:8080/money/2023/feb/20/one-in-four-vouchers-for-prepayment-meters-unredeemed-in-britain

When not supported, neither the skip links, nor the promotion should be rendered.

## Why?
We have had complaints via user-help that screen-reader users using the App have to listen to the screen reader read out the text of the promotion block on every article. This gives them an option to skip past it, making the feature less intrusive and more accessible. (They would prefer to a way to turn the promotions off globally, but we don't have instructions to do that).

## Screenshots

|focus on link|After clicking link |
|--|--|
|Screenshot 2023-02-20 at 13 02 58|Screenshot 2023-02-20 at 13 03 58  |

### CODE REVIEW

File: apps-rendering/src/components/InlineSkipToWrapper/index.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+// Using single template literals for the labels instead of
+// `after {blockDescription}` to prevent JSX rendering
+// an empty comment in the HTML between the static text
+// and the variable. That comment can make screen readers
+// announce the text oddly.
+// https://github.com/guardian/dotcom-rendering/pull/6950#pullrequestreview-1279406318
+
+const InlineSkipToWrapper: FC<Props> = ({
+	id,
```
Comment (georgeblahblah) [inline apps-rendering/src/components/InlineSkipToWrapper/index.tsx:48]: as this is being passed through to the `a`'s href / `p`'s `id`, could we strip out any invalid characters & possibly add a comment? Based on the spec, I think we need to remove any ascii whitespace (https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute)

**Review by georgeblahblah — COMMENTED** [review]
Thanks @dblatcher, this is looking good! I had one suggestion to strip out any ascii whitespace in the `id` prop to `InlineSkipToWrapper` to make sure it's a valid `id` attribute - interested to hear your thoughts

File: apps-rendering/src/components/InlineSkipToWrapper/index.tsx (hunk 1 above)
Comment (georgeblahblah) [reply apps-rendering/src/components/InlineSkipToWrapper/index.tsx:48]: actually I'm not sure a validator is really worth it, but I think a comment on the component might help so it's clear in the consumer where the value will get used

**Review by georgeblahblah — APPROVED** [review]
On reflection, I think my suggestion to add a validator is unnecessary, but I have a non-blocking suggestion to add a comment to the `InlineSkipToWrapper` component to make it clear how the `id` prop is used

**Comment by dblatcher** [comment]
> On reflection, I think my suggestion to add a validator is unnecessary, but I have a non-blocking suggestion to add a comment to the `InlineSkipToWrapper` component to make it clear how the `id` prop is used

Thanks - good idea. I've added some comments with description of the component and also on how the id is used specifically.

I also took another look at the mark-up for the `NewsletterSignup` component - I think the accessibility   tree should be clearer now with the outer container is a div with a role of "none" and the visible frame is the aside(role=complementary by default)

**Pull request merged**